.github/workflows/CD.yml

```diff
@@ -196,12 +196,21 @@ jobs:
         run: |
           if [[ -f testing-framework/terraform/testcases/${{ matrix.testcase }}/parameters.tfvars ]] ; then opts="-var-file=../testcases/${{ matrix.testcase }}/parameters.tfvars" ; else opts="" ; fi
           cd testing-framework/terraform/ec2 && terraform init && terraform apply -auto-approve -lock=false $opts -var="testing_ami=${{ matrix.testing_ami }}" -var="aoc_version=${{ needs.release-checking.outputs.version }}" -var="package_s3_bucket=aws-otel-collector" -var="testcase=../testcases/${{ matrix.testcase }}" -var="ssm_package_name=AWSDistroOTel-Collector"
-          
+
+      - name: Upload test case's terraform state
+        if: ${{ always() && steps.s3-release-validation-1.outputs.cache-hit != 'true' }}
+        run: |
+          cd testing-framework/terraform/eks
+          testing_id=$(terraform output -raw testing_id)
+          cd ../add_on/remote_state
```

Comment:
I think I must've missed the PR, but why are we using terraform to upload to S3? Wouldn't `aws s3 cp` do the same thing? Do you need to include the entire testing framework? We used to for the soaking test since it was possible to have backwards incompatible changes.

https://github.com/aws-observability/aws-otel-collector/blob/c926da59063d2173ced4094c4f58ccb335745af4/.github/workflows/soaking.yml#L189-L195

Reply:
No we don't need to include the entire workflow, only the terraform.state is fine to since we can destroy the resources by checking for the platform which the terraform.state is in. Mostly, everything should be managed by terraform for easier management and avoiding errors though.